WooCommerce/Classes/Authentication/AuthenticationManager.swift

```diff
@@ -342,18 +341,39 @@ private extension AuthenticationManager {
 
     /// Maps error codes emitted by WPAuthenticator to a domain error object
     enum AuthenticationError: Int, Error {
+        case emailDoesNotMatchWPAcount = 7
         case notWPSite = 406
         case unknown
 
         static func make(with error: Error) -> AuthenticationError {
             let error = error as NSError
 
             switch error.code {
+            case emailDoesNotMatchWPAcount.rawValue:
+                return .emailDoesNotMatchWPAcount
             case notWPSite.rawValue:
                 return .notWPSite
             default:
                 return .unknown
             }
         }
     }
+
+    func isSupportedError(_ error: Error) -> Bool {
+        let wooAuthError = AuthenticationError.make(with: error)
+        return wooAuthError == .emailDoesNotMatchWPAcount || wooAuthError == .notWPSite
```

Comment:
nit: how about just checking if the error is not `unknown`, since we handle all the known cases?
```suggestion
        return wooAuthError != .unknown
```

Reply:
I went with the long form because it seemed more explicit to me, but will update, thanks!